### DISCUSSION TITLE
Is performance usually this bad?

### DISCUSSION THREAD

**Discussion by JakeCooper** [actions/actions-runner-controller#3123] [General]
Running 0.7.0 installed via Helm. Config is super bare, just set the PAT as well as the config secret

7 seconds vs almost 3 minutes to run Golang lint?
![CleanShot 2023-12-03 at 18 47 11@2x](https://github.com/actions/actions-runner-controller/assets/5499880/73add297-a4f3-40a8-b63c-b1ccb9acbf8f)
![CleanShot 2023-12-03 at 18 47 16@2x](https://github.com/actions/actions-runner-controller/assets/5499880/2ecbf627-c2ff-4a82-882f-5695ce68f477)

I have to assume this is due to DinD + no PV but even then my goodness

**Comment by nikola-jokic** [comment]
Hey @JakeCooper,

Please correct me if I am wrong but the duration of the job execution completely depends on the environment in which the runner executes. To measure the performance of the scale set, you would have to measure the latency from the time the job is dispatched until ARC issues a create pod request to the k8s API. The rest of it is completely outside of ARC's control.
Once the job lands on the runner, the speed is completely determined by the resources allocated for the runner, network speed etc.
If you specify requests field requesting more resources, I would imagine the job would execute much faster (assuming CPU or memory is the bottleneck)

**Reply by JakeCooper** [reply]
I set it up with with a PV local SSD and that dropped it to 30 seconds for lint, which was unfortunately still 3x slower

I'll try to setup the [RAID0 PV](https://cloud.google.com/kubernetes-engine/docs/how-to/persistent-volumes/local-ssd-raw) as well as over provisioning the crap out of the workers this week
